controllers/TokenController.php

```diff
@@ -30,54 +30,58 @@ public function beforeAction($action)
 
     public function actionUpdate()
     {
-        $this->requireLogin();
+        return $this->update(false);
+    }
 
-        $driver = (new DriverService($this->module->getConfigureForm()))->getWebDriver();
-        if (!$driver) {
-            Yii::$app->response->statusCode = 400;
-            return $this->asJson(['success' => false, 'message' => 'No push driver available!']);
-        }
+    public function actionUpdateMobileApp()
+    {
+        return $this->update(true);
+    }
 
-        return $this->asJson([
-            'success' => (
-                (new TokenService())->storeTokenForUser(
-                    Yii::$app->user->getIdentity(),
-                    $driver,
-                    Yii::$app->request->post('token'),
-                )
-            ),
-        ]);
+    public function actionDelete()
+    {
+        return $this->delete(false);
     }
 
-    public function actionUpdateMobileApp()
+    public function actionDeleteMobileApp()
     {
-        $this->requireLogin();
+        return $this->delete(true);
+    }
 
-        $driver = (new DriverService($this->module->getConfigureForm()))->getMobileAppDriver();
+    private function update(bool $mobile)
```

Comment:
@marc-farre From CodeStyle I would prefer if we also pass the token to the private methods.

e.g. `$this->delete($token, $isMobileApp)`  and `$this->update($token, $isMobileApp)`. 
This makes it easier to understand.

Reply:
Good idea! Done in commit https://github.com/humhub/fcm-push/pull/63/commits/7356b594d719e102b8d0d5ef97d91ec56cd6dd63